File: src/game/game.c

#include "../shared/shared.c"

// here I'll keep a list of game projects and I can sub out which one I'm
// working on #include _game "rl2"
#include "rl/rl.c"
// #include _game "wander"

void load(App_Context* ctx) {
  if (sizeof(State) > ctx->state.size) {
    Mem old = ctx->state;
    ctx->state = (Mem){malloc(sizeof(State)), sizeof(State)};
    state = (State*)ctx->state.data;
    if (old.data != NULL) {
      unreachable;
      //   memcpy(state, old_state, old_size);
      //   free(old_state);
    }
  }
  _load();
}

void update() { _update(); }

File: src/game.zig
const std = @import("std");
const rl = @import("raylib");

// here I'll keep a list of game projects and I can sub out which one I'm
const _game = @import("game/rl/rl.zig");

const assert = std.debug.assert;
pub const State = _game.State;

pub const APP_TITLE = _game.APP_TITLE;
pub const APP_WIDTH = _game.APP_WIDTH;
pub const APP_HEIGHT = _game.APP_HEIGHT;

pub fn load() void {
    _game.load();
}

pub fn update() void {
    _game.update();
}
